extensions/amp-next-page/0.2/service.js

```diff
@@ -336,6 +351,58 @@ export class NextPageService {
     }
   }
 
+  /**
+   * Removes redundancies and unauthorized extensions and elements
+   * @param {!Document} doc Document to attach.
+   */
+  sanitizeDoc(doc) {
+    // TODO(wassgha): Parse for more pages to queue
+
+    // TODO(wassgha): Allow amp-analytics after bug bash
+    toArray(doc.querySelectorAll('amp-analytics')).forEach(removeElement);
+    // Make sure all hidden elements are initially invisible
+    this.handleElementHiding(doc, false /** isHost */, false /** isVisible */);
+  }
+
+  /**
+   * Hides or shows elements based on the `amp-next-page-hide` and
+   * `amp-next-page-replace` attributes
+   * @param {!Document} doc Document to attach.
+   * @param {boolean=} isHost Whether this is the initial page
+   * @param {boolean=} isVisible Whether this page is visible or not
+   */
+  handleElementHiding(doc, isHost = false, isVisible = true) {
```

Comment:
Let's rename the method so it's more clear what it's doing, maybe `toggleHideAndReplaceElements` or `initDocumentVisibilityStates` or similar (?)